### PR TITLE
Pub: Create ORT issue if pubspec.yaml is missing for a package

### PR DESCRIPTION
**Pull request by MarcelBochtler**
Instead of throwing an IllegalArgumentException as the yamlMapper does
not handle nullable files.

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/Pub.kt [hunk 1]
```diff
@@ -578,6 +578,16 @@ class Pub(
...
     private fun readPackageInfoFromCache(packageInfo: JsonNode): JsonNode {
         val definitionFile = reader.findFile(packageInfo, "pubspec.yaml")
+        if (definitionFile == null) {
+            createAndLogIssue(
+                source = managerName,
+                message = "Could not find 'pubspec.yaml' for '${packageInfo["name"].textValueOrEmpty()}'.",
+                severity = Severity.WARNING
+            )
+
+            return yamlMapper.createObjectNode()
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Pub.kt:588]: Maybe use our `EMPTY_JSON_NODE` instead?

File: analyzer/src/main/kotlin/managers/Pub.kt [hunk 2]
```diff
@@ -578,6 +578,16 @@ class Pub(
 
     private fun readPackageInfoFromCache(packageInfo: JsonNode): JsonNode {
         val definitionFile = reader.findFile(packageInfo, "pubspec.yaml")
+        if (definitionFile == null) {
+            createAndLogIssue(
+                source = managerName,
+                message = "Could not find 'pubspec.yaml' for '${packageInfo["name"].textValueOrEmpty()}'.",
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Pub.kt:584]: Does it make sense to move `"pubspec.yaml"` to a variable now, and also use it above?

Comment (MarcelBochtler) [reply analyzer/src/main/kotlin/managers/Pub.kt:584]: How about a `const` as this string is being used multiple times in this class?

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Pub.kt:584]: > How about a `const` as this string is being used multiple times in this class?

:+1:

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Pub.kt:586]: Now you still need to use that constant ;-)

Comment (MarcelBochtler) [reply analyzer/src/main/kotlin/managers/Pub.kt]: Done. Sorry for that :)

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**